script/03_UnificationProposal.s.sol

```diff
@@ -8,40 +8,168 @@ import {IUniswapV3Factory} from "briefcase/protocols/v3-core/interfaces/IUniswap
-    // Make governance timelock the feeToSetter for UniswapV2
-    IFeeToSetter(OLD_FEE_TO_SETTER).setFeeToSetter(timelock);
-    // Set TokenJar as the UniswapV2 fee recipient
-    V2_FACTORY.setFeeTo(address(deployer.TOKEN_JAR()));
-    // Approve 40M UNI to UNIVesting contract
-    UNI.approve(address(deployer.UNI_VESTING()), 40_000_000 ether);
+    // --- Proposal Actions Setup ---
+    address[] memory targets = new address[](5);
+    uint256[] memory values = new uint256[](5);
+    string[] memory signatures = new string[](5);
+    bytes[] memory calldatas = new bytes[](5);
+
+    // Action 0: Burn 100M UNI
+    targets[0] = address(UNI);
+    values[0] = 0;
+    signatures[0] = "";
+    calldatas[0] = abi.encodeCall(UNI.transfer, (address(0xdead), 100_000_000 ether));
+
+    // Action 1: Enable UniswapV3 FeeAdapter
+    targets[1] = address(V3_FACTORY);
+    values[1] = 0;
+    signatures[1] = "";
+    calldatas[1] = abi.encodeCall(V3_FACTORY.setOwner, (address(deployer.V3_FEE_ADAPTER())));
+
+    // Action 2: Make governance timelock the feeToSetter for UniswapV2
+    targets[2] = address(OLD_FEE_TO_SETTER);
+    values[2] = 0;
+    signatures[2] = "";
+    calldatas[2] = abi.encodeCall(IFeeToSetter.setFeeToSetter, (timelock));
+
+    // Action 3: Set TokenJar as the UniswapV2 fee recipient
+    targets[3] = address(V2_FACTORY);
+    values[3] = 0;
+    signatures[3] = "";
+    calldatas[3] = abi.encodeCall(V2_FACTORY.setFeeTo, (address(deployer.TOKEN_JAR())));
+
+    // Action 4: Approve 40M UNI to UNIVesting contract
+    targets[4] = address(UNI);
+    values[4] = 0;
+    signatures[4] = "";
+    calldatas[4] = abi.encodeCall(UNI.approve, (address(deployer.UNI_VESTING()), 40_000_000 ether));
+
+    // Action 5: Attest to Agreement 1
+    if (AGREEMENT_ANCHOR_1 != address(0)) {
+      targets[5] = address(EAS);
+      values[5] = 0;
+      signatures[5] = "";
+      calldatas[5] = abi.encodeCall(
+        EAS.attest,
+        (AttestationRequest({
+            schema: AGREEMENT_SCHEMA_UID,
+            data: AttestationRequestData({
+              recipient: AGREEMENT_ANCHOR_1,
+              expirationTime: 0,
+              revocable: false,
+              refUID: bytes32(0),
+              data: abi.encode(CONTENT_HASH_1),
+              value: 0
+            })
+          }))
+      );
+    }
+
+    // Action 6: Attest to Agreement 2
+    if (AGREEMENT_ANCHOR_2 != address(0)) {
+      targets[6] = address(EAS);
+      values[6] = 0;
+      signatures[6] = "";
+      calldatas[6] = abi.encodeCall(
+        EAS.attest,
+        (AttestationRequest({
+            schema: AGREEMENT_SCHEMA_UID,
+            data: AttestationRequestData({
+              recipient: AGREEMENT_ANCHOR_2,
+              expirationTime: 0,
+              revocable: false,
+              refUID: bytes32(0),
+              data: abi.encode(CONTENT_HASH_2),
+              value: 0
+            })
+          }))
+      );
+    }
```

Comment:
This looks good - I would maybe suggest though adding some basic asserts to the new owner fields on v3 & v2 esp since we will know the addresses.